mediagrains/testsignalgenerator.py

```diff
@@ -124,6 +124,127 @@ def LumaSteps(src_id, flow_id, width, height,
         vg.sync_timestamp = vg.origin_timestamp
 
 
+
+def ColourBars(src_id, flow_id, width, height,
+               intensity=0.75,
+               rate=Fraction(25, 1),
+               origin_timestamp=None,
+               cog_frame_format=CogFrameFormat.U8_444,
+               step=1):
+    """Returns a generator for colour bar video grains in specified format.
+    :param src_id: source_id for grains
+    :param flow_id: flow_id for grains
+    :param width: width of grains
+    :param height: height of grains
+    :param intensity: intensity of colour bars (usually 1.0 or 0.75)
+    :param rate: rate of grains
+    :param origin_timestamp: the origin timestamp of the first grain.
+    :param step: The number of grains to increment by each time (values above 1 cause skipping)"""
+
+    if cog_frame_format not in pixel_ranges:
+        raise ValueError("Not a supported format for this generator")
+
+    _bpp = pixel_ranges[cog_frame_format][0]
+    _steps = 8
+    bs = 16 - pixel_ranges[cog_frame_format][4]
+
+    values = [
```

Comment:
I notice that some numbers below pad out using 0xFF and some 0x00, e.g. 0xFFFF and 0x8000. Is there are reason for that? I'm comparing the numbers with those in videoio which are 8-bit. It might be that these >8-bit paddings might be stated elsewhere in a spec for example.

Reply:
I padded by extending the lowest bit. Arguably not the correct solution but not a bad one, I think.